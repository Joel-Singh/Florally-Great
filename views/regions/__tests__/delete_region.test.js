const renderPugToDOM = require('./util/renderPugToDOM.js')

test('renders single label and input with one region', () => {
  const renderedHtml = renderPugToDOM(
    './views/regions/delete_region.pug',
    {all_regions: [{name: "RegionName", _id: 1234}]}
  )

  expect(renderedHtml.querySelector('form')).toMatchSnapshot()
})

test('render multiple labels and inputs', () => {
  const all_regions = [
    {
      name: "A",
      _id: 1
    },
    {
      name: "B",
      _id: 2
    },
    {
      name: "C",
      _id: 3
    }
  ]

  const renderedHtml = renderPugToDOM(
    './views/regions/delete_region.pug',
    { all_regions }
  )

  expect(renderedHtml.querySelector('form')).toMatchSnapshot()
})
