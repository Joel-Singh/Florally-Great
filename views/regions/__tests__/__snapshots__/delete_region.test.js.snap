// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render multiple labels and inputs 1`] = `
<form
  action=""
  method="POST"
>
  <label
    for="A"
  >
    A
  </label>
  <input
    name="region"
    type="radio"
    value="1"
  />
  <label
    for="B"
  >
    B
  </label>
  <input
    name="region"
    type="radio"
    value="2"
  />
  <label
    for="C"
  >
    C
  </label>
  <input
    name="region"
    type="radio"
    value="3"
  />
  <button
    type="submit"
  >
    Delete
  </button>
</form>
`;

exports[`renders single label and input with one region 1`] = `
<form
  action=""
  method="POST"
>
  <label
    for="RegionName"
  >
    RegionName
  </label>
  <input
    name="region"
    type="radio"
    value="1234"
  />
  <button
    type="submit"
  >
    Delete
  </button>
</form>
`;
